File: AspNetCore/TestResource.Api/Controllers/ExtractController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Net;
using System.Threading.Tasks;
using TestResource.Domain.IApplicationService;
using TestResource.Domain.Model;
using TestResource.Domain.ViewModel;

namespace TestResource.Api.Controllers
{
    [Route("Extract")]
    public class ExtractController : BaseController
    {
        private readonly IExtractApplicationService _application;

        public ExtractController(IExtractApplicationService application)
        {
            _application = application;
        }

        [HttpGet]
        [ProducesResponseType(typeof(BaseReturn<List<ExtractViewModel>>), (int)HttpStatusCode.OK)]
        [ProducesResponseType(typeof(BaseReturn<object>), (int)HttpStatusCode.InternalServerError)]
        public async Task<IActionResult> List()
        {
            var response = new BaseReturn<object>();

            try
            {
                var service = await _application.List();

                response.Data = service.Data;
            }
            catch (Exception ex)
            {
                response.Success = false;
                response.Description = ex.Message;
            }

            return HandleResponse(response);
        }
    }
}

File: Spring/src/main/java/com/test/controller/ExtractController.java
package com.test.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.List;

import com.test.model.BaseReturn;
import com.test.viewModel.ExtractViewModel;
import com.test.service.ExtractService;

@RestController
public class ExtractController {

    @Autowired
    private ExtractService _service;

    public ExtractController(ExtractService service) {
        this._service = service;
    }

    @GetMapping("/")
    public BaseReturn<List<ExtractViewModel>> List() {
        BaseReturn<List<ExtractViewModel>> response = new BaseReturn<List<ExtractViewModel>>();

        try {

            BaseReturn<List<ExtractViewModel>> viewModels = _service.List();

            response.Data = viewModels.Data;
        } catch (Exception ex) {
            response.Success = false;
            response.Description = ex.getMessage();
        }

        return response;
    }
}